include/fmt/format.h

```diff
@@ -2112,8 +2112,14 @@ FMT_CONSTEXPR OutputIt write(OutputIt out, T value) {
 }
 
 template <typename Char, typename OutputIt, typename T,
+#ifdef __cpp_lib_byte
+          FMT_ENABLE_IF(std::is_enum<T>::value &&
+                        !std::is_same<T, Char>::value &&
+                        !std::is_same<T, std::byte>::value)>
+#else
           FMT_ENABLE_IF(std::is_enum<T>::value &&
                         !std::is_same<T, Char>::value)>
+#endif
```

Comment:
This should have an additional SFINAE check `mapped_type_constant<T, basic_format_context<OutputIt, Char>>::value != type::custom_type` instead of the check if T is byte.